### PR TITLE
Fixing fault view state management

### PR DESCRIPTION
**Pull request by itamarsch**

### CODE REVIEW

**Comment by itamarsch** [comment]
@giladwo Turns out we don't really need the checkboxes so all of the state manegement is pretty easy and we don't need two stateful widgets

**Comment by giladwo** [comment]
> @giladwo Turns out we don't really need the checkboxes so all of the state management is pretty easy and we don't need two stateful widgets

Good, reducing state is always a good move.

**Comment by itamarsch** [comment]
@giladwo Do you approve how I did the state manegament here?

File: lib/views/mobile/screens/fault_view.dart [hunk 1]
```diff
@@ -3,7 +3,7 @@ import "package:graphql/client.dart";
 import "package:scouting_frontend/models/map_nullable.dart";
 import "package:scouting_frontend/models/team_model.dart";
 import "package:scouting_frontend/net/hasura_helper.dart";
-import "package:scouting_frontend/views/mobile/fault_view_list.dart";
+import "package:scouting_frontend/views/constants.dart";
 import "package:scouting_frontend/views/mobile/side_nav_bar.dart";
 
 class FaultView extends StatefulWidget {
```
Comment (giladwo) [inline lib/views/mobile/screens/fault_view.dart:9]: Why is it still a `StatefulWidget`?

Comment (itamarsch) [reply lib/views/mobile/screens/fault_view.dart:9]: Because there is state the lives on the db of which team has a fault

Comment (giladwo) [reply lib/views/mobile/screens/fault_view.dart:9]: This doesn't require a `StatefulWidget` though.
You don't have any state here...

Comment (itamarsch) [reply lib/views/mobile/screens/fault_view.dart:9]: So how do I rerender the widget when it's stateless

Comment (giladwo) [reply lib/views/mobile/screens/fault_view.dart:9]: `_FaultViewState` doesn't hold any state, so why should it be re-rendered?

Comment (itamarsch) [reply lib/views/mobile/screens/fault_view.dart:9]: Because I want future builder to requery

**Review by giladwo — APPROVED** [review]

**Comment by giladwo** [comment]
Note that using subscriptions avoids this empty `setState` smell.

File: lib/views/mobile/screens/fault_view.dart (hunk 1 above)
Comment (giladwo) [reply lib/views/mobile/screens/fault_view.dart:9]: @itamarsch Note that using subscriptions avoids this empty `setState` smell.

**Pull request merged**